File: app/src/main/java/com/osrsd/cache/def/SpotAnimationDefinition.java
package com.osrsd.cache.def;

import lombok.Data;

@Data
public final class SpotAnimationDefinition implements Definition {

    private final int id;
    private int rotation;
    private short[] textureToReplace;
    private short[] textureToFind;
    private int resizeY;
    private int animationId;
    private short[] recolorToFind;
    private short[] recolorToReplace;
    private int resizeX;
    private int modelId;
    private int ambient;
    private int contrast;

    public SpotAnimationDefinition(int id) {
        this.id = id;
        rotation = 0;
        resizeY = 128;
        animationId = -1;
        resizeX = 128;
        ambient = 0;
        contrast = 0;
    }

}

File: app/src/main/kotlin/com/osedit/spring/domain/SpotAnimationDefinition.kt
package com.osedit.spring.domain

import com.osedit.cache.Config
import com.osedit.cache.Indices
import com.osedit.cache.provider.SpotAnimationProvider
import java.nio.ByteBuffer
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class SpotAnimationDefinition(@Id private val id: Int? = 0) : Definition {
    var rotation: Int = 0
    var textureToReplace: ShortArray? = null
    var textureToFind: ShortArray? = null
    var resizeY: Int = 128
    var animationId: Int = -1
    var recolorToFind: ShortArray? = null
    var recolorToReplace: ShortArray? = null
    var resizeX: Int = 128
    var modelId: Int = 0
    var ambient: Int = 0
    var contrast: Int = 0

    override fun path(): String {
        return "/spotanimations"
    }

    override fun config(): Config {
        return Config.SPOT_ANIMS
    }

    override fun indices(): Indices {
        return Indices.CONFIG
    }

    override fun decode(byteBuffer: ByteBuffer): Definition {
        return SpotAnimationProvider.decode(byteBuffer, this)
    }
}